torch/quantization/ns/numeric_suite_core_apis_fx.py

```diff
@@ -27,65 +27,71 @@
 
-    for match_name, match in matched_subgraph_pairs.items():
-
-        (node_start_a, node_end_a), (node_start_b, node_end_b) = match
-        # Note: for matching weights we always use the start nodes,
-        # such as using linear in linear-relu
-        assert node_start_a.op == node_start_b.op and \
-            node_start_a.op in ('call_function', 'call_module')
+        if ref_node_name not in results:
+            results[ref_node_name] = {}
 
-        if node_start_a.op == 'call_function':
+        if node.op == 'call_function':
 
             # linear
             # TODO(future PR): other function types
-            a_related_to_linear = node_start_a.target in (F.linear,) or \
-                (node_start_a.target, F.linear) in type_a_related_to_b
+            related_to_linear = node.target in (F.linear,) or \
+                (node.target, F.linear) in type_a_related_to_b
 
-            if a_related_to_linear:
-                weight_a = get_linear_fun_weight(node_start_a, gm_a)
-                weight_b = get_linear_fun_weight(node_start_b, gm_b)
-
-                results[match_name] = {
-                    name_a: weight_a,
-                    name_b: weight_b,
-                }
+            if related_to_linear:
+                weight = get_linear_fun_weight(node, model)
+                results[ref_node_name][model_name] = weight
 
         else:  # call_module
             # for call_module, we need to look up the modules to do the type check
-            assert isinstance(node_start_a.target, str)
-            mod_a = getattr_from_fqn(gm_a, node_start_a.target)
-            assert isinstance(node_start_b.target, str)
-            mod_b = getattr_from_fqn(gm_b, node_start_b.target)
+            assert isinstance(node.target, str)
+            mod = getattr_from_fqn(model, node.target)
 
             # check that A is one the modules we need
             # assume B is related (this is done by graph matcher)
-            a_related_to_conv2d_mod = isinstance(mod_a, nn.Conv2d) or \
-                (type(mod_a), nn.Conv2d) in type_a_related_to_b
+            related_to_conv2d_mod = isinstance(mod, nn.Conv2d) or \
+                (type(mod), nn.Conv2d) in type_a_related_to_b
 
             # TODO(future PR): other module types
-            if a_related_to_conv2d_mod:
-                weight_a = get_conv_mod_weight(mod_a)
-                weight_b = get_conv_mod_weight(mod_b)
-                results[match_name] = {
-                    name_a: weight_a,
-                    name_b: weight_b,
-                }
+            if related_to_conv2d_mod:
+                weight = get_conv_mod_weight(mod)
+                results[ref_node_name][model_name] = weight
+
+# Note: this is not a user facing API
+# TODO(future PR): wrap this in a user facing API which does not
+#   expose FX types.
+def compare_weights(
+    model_name_a: str,
+    gm_a: GraphModule,
+    model_name_b: str,
+    gm_b: GraphModule,
```

Comment:
For a future PR: We need to provide flexibility to users in terms of specifying modules that they want to compare both in terms of module names and in terms of module types.

Reply:
agreed, saving that for a future PR